seed/seed20_test.go

```diff
@@ -2873,3 +2876,183 @@ func (s *seed20Suite) TestLoadMetaWrongHashSnapParallelism2(c *C) {
 	err = seed20.LoadMeta(seed.AllModes, nil, s.perfTimings)
 	c.Check(err, ErrorMatches, `cannot validate ".*pc-kernel_1\.snap" for snap "pc-kernel" \(snap-id "pckernel.*"\), hash mismatch with snap-revision`)
 }
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeSecuredNoAutoImportAssertion(c *C) {
+	// secured grade, no su assertion
+	s.testLoadAutoImportAssertion(c, asserts.ModelSecured, none, 0644, s.commitTo)
+}
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeSecuredAutoImportAssertion(c *C) {
+	// secured grade, with su assertion
+	s.testLoadAutoImportAssertion(c, asserts.ModelSecured, valid, 0644, s.commitTo)
+}
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeDangerousNoAutoImportAssertion(c *C) {
+	// dangerous grade, no su assertion
+	s.testLoadAutoImportAssertion(c, asserts.ModelDangerous, none, 0644, s.commitTo)
+}
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeDangerousAutoImportAssertionErrCommiter(c *C) {
+	// dangerous grade with broken commiter
+	s.testLoadAutoImportAssertion(c, asserts.ModelDangerous, valid, 0644, func(b *asserts.Batch) error {
+		return fmt.Errorf("nope")
+	})
+}
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeDangerousAutoImportAssertionErrFilePerm(c *C) {
+	// dangerous grade, su assertion with wrong file permissions
+	s.testLoadAutoImportAssertion(c, asserts.ModelDangerous, valid, 0222, s.commitTo)
+}
+
+func (s *seed20Suite) TestLoadAutoImportAssertionGradeDangerousInvalidAutoImportAssertion(c *C) {
+	// dangerous grade, invalid su assertion
+	s.testLoadAutoImportAssertion(c, asserts.ModelDangerous, invalid, 0644, s.commitTo)
+}
+
+type suAsseertion int
+
+const (
+	none suAsseertion = iota
+	valid
+	invalid
+)
+
+func (s *seed20Suite) testLoadAutoImportAssertion(c *C, grade asserts.ModelGrade, sua suAsseertion, perm os.FileMode, commitTo func(b *asserts.Batch) error) {
+	sysLabel := "20191018"
+	seed20 := s.createMinimalSeed(c, string(grade), sysLabel)
+	c.Assert(seed20, NotNil)
+	c.Check(seed20.Model().Grade(), check.Equals, grade)
+
+	// write test auto import assertion
+	switch sua {
+	case valid:
+		s.writeValidAutoImportAssertion(c, sysLabel, perm)
+	case invalid:
+		s.writeInvalidAutoImportAssertion(c, sysLabel, perm)
+	}
+
+	// try to load auto import assertions
+	err := seed20.LoadAutoImportAssertion(commitTo)
+
+	c.Assert(err, IsNil)
+	assertions, err := s.findAutoImportAssertion(seed20)
+	c.Assert(err, NotNil)
```

Comment:
This is not needed, given the check below

Reply:
Yep I will rename this as separate PR